library/src/hcc_detail/hipsolver.cpp

```diff
@@ -306,20 +306,24 @@ hipsolverStatus_t hipsolverSorgbr_bufferSize(hipsolverHandle_t   handle,
                                              int*                lwork)
 try
 {
+    if(lwork == nullptr)
+        return HIPSOLVER_STATUS_INVALID_VALUE;
+
+    *lwork = 0;
```

Comment:
I'm curious why we're setting this to zero when there's an error. Is that to match existing CUDA or LAPACK behaviour?

Reply:
This is to address the test failure I was getting from gesvd. Since we don't initialize `size_W` or `lwork` or check for bufferSize errors in the tests, `lwork` was taking a random (large) value and we were getting an allocation errors. Alternatively, I could have changed something in the tests, but I thought this would increase the safety of our code.

Reply:
If CUDA doesn't guarantee this, won't it be a problem in when we run the tests on the CUDA build?

Reply:
Oh, good point! I'll check what CUDA does. If they don't set lwork in the case of an error, I'll update the tests so that `size_W` is initialized to zero.

Reply:
It looks like CUDA does set `lwork` even if the problem size is invalid (for gesvd, at least). Interestingly, it's a non-zero value.

Reply:
That is rather odd. I feel like it would be better if the tests checked for errors, but we can probably merge as-is.